Description: Disable installation of JSmol nbextension
 Since we are having difficulty packaging that
 .
 The other things (e.g. notebook iframes) don't cause doctest failures and
 we already documented it in README.Debian
Author: Ximin Luo <infinity0@debian.org>
Forwarded: not-needed
--- a/sage/src/sage/repl/ipython_kernel/install.py
+++ b/sage/src/sage/repl/ipython_kernel/install.py
@@ -121,23 +121,6 @@
         dst = os.path.join(self.nbextensions_dir, 'mathjax')
         self.symlink(src, dst)
 
-    def use_local_jsmol(self):
-        """
-        Symlink jsmol to the Jupyter notebook.
-
-        EXAMPLES::
-
-            sage: from sage.repl.ipython_kernel.install import SageKernelSpec
-            sage: spec = SageKernelSpec()
-            sage: spec.use_local_jsmol()
-            sage: jsmol = os.path.join(spec.nbextensions_dir, 'jsmol')
-            sage: os.path.isdir(jsmol)
-            True
-        """
-        src = os.path.join(SAGE_LOCAL, 'share', 'jsmol')
-        dst = os.path.join(self.nbextensions_dir, 'jsmol')
-        self.symlink(src, dst)
-
     def use_local_threejs(self):
         """
         Symlink threejs to the Jupyter notebook.
@@ -260,7 +243,6 @@
         """
         instance = cls()
         instance.use_local_mathjax()
-        instance.use_local_jsmol()
         instance.use_local_threejs()
         instance._install_spec()
         instance._symlink_resources()
--- a/sage/src/sage/repl/display/jsmol_iframe.py
+++ b/sage/src/sage/repl/display/jsmol_iframe.py
@@ -42,10 +42,23 @@
       height: 100%;
       width: 100%;
     }}
+    ul.debian-jsmol {{
+      list-style-type: disc;
+      list-style-position: inside;
+    }}
+    ul.debian-jsmol li {{
+      margin: 1em 0;
+    }}
   </style>
   <script type="text/javascript" src="{path_to_jsmol}/JSmol.min.js"></script>
 </head>
 <body>
+  <p>Interactive 3D plots with the default viewer do not work in the notebook, because JSmol cannot currently be packaged for Debian. See the README.source of the jmol package for more details on this situation. You have the following alternative options:</p>
+  <ul class="debian-jsmol">
+  <li>View an interactive WebGL-based plot by giving the option <code>viewer='threejs'</code>. This should work out-of-the box on Firefox, but may require you to adjust some settings in Chromium - see <code>chrome://flags</code> for options and <code>chrome://gpu</code> for status. In both cases, the functionality also depends on your graphics hardware and drivers, but support should exist in most cases.</li>
+  <li>View a non-interactive plot by giving the option <code>viewer='tachyon'</code>.</li>
+  <li>Use the default viewer (Jmol) from the SageMath command-line interface, instead of the notebook.</li>
+  </ul>
   <script type="text/javascript">
     var script = {script};
     var Info = {{
